lib/src/linting/handle-lint-violations.ts

```diff
@@ -0,0 +1,69 @@
+import { GroupedLintRuleViolations } from "./rule";
+import { SpotConfig } from "cli/src/commands/lint";
+
+/**
+ * Responsible for triggering error or warn depending on whether the lint rule
+ * violation setting is 'off', 'warn' or 'error'.
+ *
+ * By default, if a lint rule setting is not set in spotConfig,
+ * then it will be considered a error.
+ *
+ * Returns the deferExit value which will be true if there is a lint violation
+ * error or an unknown rule setting is found. Otherwise it is false.
+ */
+interface HandleLintViolationsDependencies {
+  error: (
+    msg: string,
+    options: {
+      code?: string;
+      exit: false;
+    }
+  ) => void;
+  warn: (msg: string) => void;
+}
+
+export const handleLintViolations = (
+  groupedLintErrors: GroupedLintRuleViolations[],
+  spotConfig: SpotConfig,
+  { error, warn }: HandleLintViolationsDependencies
+): boolean => {
```

Comment:
Instead of returning a `boolean`, which means you always need to read the JSDoc to understand what it will represent, how about returning a more explicit object? I'm thinking `{ foundErrors: boolean }` for example. You could actually make this a bit more general such as `{ errorCount: number, warningCount: number }`.

This also means that the responsibility of deciding whether to exit or not moves onto the caller.

Reply:
Yeah, I like your suggestion of `{ errorCount: number, warningCount: number }`.
Could potentially add something like  `max-warnings` in the future with that